website/content/en/docs/advanced-topics/heterogeneous.md

```diff
@@ -0,0 +1,400 @@
+---
+title: Heterogeneous Cluster Support
+linkTitle: Heterogeneous Cluster Support
+weight: 80
+---
+## Overview
+A heterogeneous cluster is a cluster in which the nodes differ from each other in a way that makes their containers incompatible with each other. The most common incompatibility that applies is CPU architecture. The rest of this document assumes that the user is developing an application that needs to be able to deploy cleanly in a cluster with a mix of node CPU architectures. Because an operator deployed to a heterogeneous cluster may not support all of the architectures in the cluster, it is important to set up safeguards that prevent scheduling pods on nodes of incompatible architectures. These scheduling implications affect both the operator as well as its operands.
+
+### Determining the Supported Architectures for an Operator/Operand Image
+For a given container image, you can check which platforms are supported by listing them by inspecting the manifest.
+```bash
+docker manifest inspect <image>
+```
+If the above command fails, it may be because the referenced name points to a single image rather than a manifest list. In this case, you can find the architecture by inspecting the image.
+```bash
+docker inspect <image>
+```
+
+Here's an example of the architectures distributed for the Alpine Linux container image.
+```bash
+$ docker manifest inspect alpine | grep architecture
+            "architecture": "amd64",
+            "architecture": "arm",
+            "architecture": "arm",
+            "architecture": "arm64",
+            "architecture": "386",
+            "architecture": "ppc64le",
+            "architecture": "s390x",
+```
+
+#### Shipping a Multi-Architecture Operator
+If once you've determined which subset of architecture nodes you want to distribute your operator for, there are tools available to help you distribute your operator.
+
+OperatorSDK provides a validator to help check bundle configuration.
```

Comment:
heterogeneous and multiple arch is the same thing right?
It seems like it is not by reading the doc

IHMO speaking about the validator here is confusing because I think that we should explain how things work and what are the pre-requirements and recommendations. After all, we can say something like:

 if you are integrating your operator with OLM then you will create a bundle, the bundle will be created based on the configuration provided in the config/ directory. To help you out validate the multiple architecture configuration you can try to test your bundle with operator-sdk bundle validate ./bundle --select-optional name=multiarch.

However, note that all workloads created in your project must be configured accordingly as well and the above validator is capable of checking only what is configured in the bundle CSV manifest.

Reply:
Heterogeneous and multi-arch do not typically mean the same. Thing thing.

All clusters that have nodes of different architectures are heterogeneous.
Not all heterogeneous clusters involve nodes of different architectures. For example, you mentioned "darwin/arm64" vs "linux/arm64". Theoretically, you could have a homogeneous architecture cluster that is heterogeneous by operating system.

Finally, when we talk about multi-arch (as a general rule), we are usually talking about homogeneous clusters on a architecture other than amd64. That is - all nodes are uni-formally arm64, ppc64le, or s390x. It is only recently that we've extended the multi-architecture scope to include non-homogeneous environments.

--

With specific regard to the validator - I was intending on following up this patch with either:
1. a heterogeneous validator
2. an extension to the multi-arch validator to include heterogeneous considerations

So while the discussion is incomplete, I do think it's important to highlight was is already available to help start this journey. To be clear, talking about heterogeneous support generally isn't relevant until the cluster already support multiple architectures.